frontend/src/app/accountCreation/AccountCreationApp.tsx

```diff
@@ -22,67 +26,77 @@ import { MfaPhoneVerify } from "./MfaPhoneVerify/MfaPhoneVerify";
 import { MfaOktaVerify } from "./MfaOktaVerify/MfaOktaVerify";
 import { MfaGoogleAuthVerify } from "./MfaGoogleAuthVerify/MfaGoogleAuthVerify";
 import { PasswordForm } from "./PasswordForm/PasswordForm";
-
-interface WrapperProps {
-  activationToken: string;
-}
-const AccountCreation404Wrapper: FunctionComponent<WrapperProps> = ({
-  activationToken,
-  children,
-}) => {
-  if (activationToken === undefined) {
-    return <>Loading...</>;
-  }
-  if (activationToken === null) {
-    return <PageNotFound />;
-  }
-  return <>{children}</>;
-};
+import { AccountCreationApi } from "./AccountCreationApiService";
+import { routeFromStatus, UserAccountStatus } from "./UserAccountStatus";
 
 const AccountCreationApp = () => {
-  const dispatch = useDispatch();
-  const activationToken = useSelector<RootState, string>(
-    (state) => state.activationToken
+  // Initialize to loading state on app load
+  const [userAccountStatus, setUserAccountStatus] = useState(
+    UserAccountStatus.LOADING
   );
+  // Used to reroute based on user's status
+  const history = useHistory();
 
+  // Runs once on app load
   useEffect(() => {
-    dispatch(
-      setInitialState({
-        activationToken: getActivationTokenFromUrl(),
-      })
-    );
-  }, [dispatch]);
+    const getStatusAndActivate = async (
+      activationToken: string | null = null
+    ) => {
+      // Ask backend what the user's current account creation status is
+      let status = await AccountCreationApi.getUserStatus(activationToken);
+      // If the user hasn't been activated yet w/ the activation token, do so
+      if (status === UserAccountStatus.PENDING_ACTIVATION && activationToken) {
+        await AccountCreationApi.initialize(activationToken);
+        // Re-retrieve the status since it will have changed after activation
+        status = await AccountCreationApi.getUserStatus();
+      }
+      // Get correct path based on status
+      const newPath = routeFromStatus(status);
+      // Check what current path is
+      const mount = history.location.pathname.split("/uac")[0];
+      const currentPath = history.location.pathname.split("/uac")[1];
+      // If new path is different, reroute
+      if (newPath !== currentPath) {
+        history.push(`${mount}/uac${newPath}`);
+      }
+      // Set the userAccountStatus state, triggering a rerender w/ the Router
+      setUserAccountStatus(status);
+    };
+    const token = getActivationTokenFromUrl();
+    getStatusAndActivate(token);
```

Comment:
I think so, but @nickclyde wrote this piece so I'm not entirely sure. I suspect it is intentional because if a new route is pushed we need to check the user's status to see if they can visit that page.

Reply:
Actually, no, that's not intentional. I only want this effect to run once on app load - but since I use `history` in it, I was forced to add it as a dependency. Is there any way to get this to only run once as intended? :thinking: